daml_dit_if/main/common.py

```diff
@@ -85,10 +85,13 @@ async def wrapped(*args, **kwargs):
                 )
 
                 inv_status.command_count += len(response.commands)
-
-                await wait_for(
-                    client.submit(response.commands), response.command_timeout
-                )
+                try:
+                    await wait_for(
+                        client.submit(response.commands), response.command_timeout
+                    )
+                except Exception as e:
+                    if response.error_handler:
+                        await response.error_handler(e)
```

Comment:
I think I would do the call itself outside the actual `catch` block, and in its own `try` (with separate logging, but no additional call to `error_handler`. This is on the general premise that if the exception handler fails, then things are probably too far gone for the IF to attempt any further recovery.